### PR TITLE
 Add --canonical-base-url cli option to docs generator

### LINKED ISSUE #5952: Outdated API doc links in Google search (bad SEO)
For example, https://www.google.com/search?q=crystal+lang+namedtuple
finds https://crystal-lang.org/api/0.21.1/NamedTuple.html
(current version is 0.24.2).

This is bad because

1) people see old docs
2) the links from different versions fight each other for dominance instead of joining forces.

Python had this problem for a while, they seem to have solved it by adding a `canonical` ref to all their pages. Perhaps technically it's not the intended use of this tag, but it has definitely worked.
For example, https://www.google.com/search?q=python+socket+doc
finds the "latest" page
because https://docs.python.org/3.5/library/socket.html
contains `<link rel="canonical" href="https://docs.python.org/3/library/socket.html" />`

Proposed solution: edit every existing Crystal doc page in storage and add this tag.
For example, 
https://crystal-lang.org/api/0.21.1/BigInt.html
```diff
--- a/BigInt.html
+++ b/BigInt.html
@@ -6,6 +6,7 @@
   <link href="css/style.css" rel="stylesheet" type="text/css" />
   <script type="text/javascript" src="js/doc.js"></script>
   <title>BigInt - github.com/crystal-lang/crystal</title>
+  <link rel="canonical" href="https://crystal-lang.org/api/latest/BigInt.html" />
 </head>
 <body>
```

I'm not entirely sure if this will work the same, because in Python the `/latest/` (`/3/`) page actually exists as an alias, and is not a redirect. So maybe that would need to be changed as well.

### PR DESCRIPTION
**Pull request by Sija**
Closes #5952

### CODE REVIEW

**Comment by straight-shoota** [comment]
This won't work: The docs generator is not only used for the standard lib. With this change, any API docs would point to `https://crystal-lang.org/api/latest/` as canonical URL!

This could probably be implemented as an optional CLI flag and entirely omitted unless explicitly specified.

And `CANONICAL_BASE_URL` would be a more descriptive name.

**Comment by Sija** [comment]
My bad, I made a mental shortcut 😅

**Comment by Sija** [comment]
Now it'll add it only for crystal repo docs. It's a kinda dirty hack, maybe cli option would be better... WDYT?

File: src/compiler/crystal/tools/doc/html/main.html [hunk 1]
```diff
@@ -5,8 +5,11 @@
   <meta id="repository-name" content="<%= repository_name %>">
   <title>README - <%= repository_name %></title>
   <script type="text/javascript">
-  CrystalDoc.base_path = "";
+    CrystalDoc.base_path = "";
   </script>
+  <% if repository_name == CRYSTAL_REPOSITORY_NAME %>
+    <link rel="canonical" href="<%= CANONICAL_BASE_URL %>" />
+  <% end %>
```
Comment (oprypin) [inline src/compiler/crystal/tools/doc/html/main.html]: Why not add this to `HeadTemplate`?
src/compiler/crystal/tools/doc/html/_head.html

**Comment by oprypin** [comment]
Before merging this, someone with access should do [the experiment](https://github.com/crystal-lang/crystal/issues/5952#issuecomment-381439285) and later change all old pages.

File: src/compiler/crystal/tools/doc/html/main.html (hunk 1 above)
Comment (Sija) [reply src/compiler/crystal/tools/doc/html/main.html]: AFAIK there's no `type` in `HeadTemplate`.

**Comment by straight-shoota** [comment]
Please just make it configurable. This feature is useful for other crystal libraries as well.

**Comment by Sija** [comment]
@straight-shoota done.

**Comment by straight-shoota** [comment]
Maybe it would be better to put it in the HeadTemplate (add type), along with that repository name and base path, too.

**Comment by Sija** [comment]
@straight-shoota done.

**Comment by Sija** [comment]
@RX14 @sdogruyol ready for review.

**Pull request merged**